src/ros/mb_publisher.cpp

```diff
@@ -16,12 +16,30 @@ void MBPublisher::publish(ur_msgs::IOStates& io_msg, SharedMasterBoardData& data
   appendAnalog(io_msg.analog_out_states, data.analog_output1, 1);
 
   io_pub_.publish(io_msg);
+
+  // --- Publishes the data in robot-specific format.
+  ur_msgs::MasterboardDataMsg msg;
+  msg.digital_input_bits = 0;   // TODO: Set this.
+  msg.digital_output_bits = 0;  // TODO: Set this.
```

Comment:
There's also this, which is not straightforward to fix and not sure it's worth the effort.

Not straightforward because properly setting this would require changing the `MasterboardDataMsg` to accomodate 18 bits in these fields.

Not worth the effort because this information is already published in the `IOStates` messages.

Reply:
I'm confused: aren't the bits just part of a bitmask in the original data (according to the protocol spec these fields are 4 byte integers)? If so: these fields should just be set to the value in the original message.

@miguelprada: are you suggesting that each individual pin should be addressable in `digital_input_bits` (ie: make it a list or something like that)?

Reply:
To be honest, I haven't looked much into this. Even if the protocol marks them as integers, I reckon that these are parsed into bitmasks of size [10](https://github.com/ros-industrial/ur_modern_driver/blob/093e1fbd2a6f81286f318aad20f539b5eb494259/include/ur_modern_driver/ur/master_board.h#L63-L64) or [18](https://github.com/ros-industrial/ur_modern_driver/blob/093e1fbd2a6f81286f318aad20f539b5eb494259/include/ur_modern_driver/ur/master_board.h#L84-L85) for a reason.

Those 18 bits not fitting in the `MasteRboarDataMsg` was why I said it wasn't straightforward to fix. I saw you are already trying to fix it, though :wink: